.github/workflows/test.yml

```diff
@@ -6,35 +6,53 @@ on:
   pull_request:
 
 jobs:
-  Go:
+  go-test:
+    name: Go Test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
       with:
         submodules: recursive
 
     - name: Set up Go
-      uses: actions/setup-go@v3
+      uses: actions/setup-go@v4
       with:
-        go-version: '1.20'
+        go-version-file: go.mod
```

Comment:
oh cool trick! Wondering if we can do the same with goreleaser instead of having the .go-version file.